driver-core/src/main/com/mongodb/client/model/expressions/IntegerExpression.java

```diff
@@ -20,5 +20,26 @@
  * Expresses an integer value.
  */
 public interface IntegerExpression extends NumberExpression {
```

Comment:
They are on integerExpression specifically because `intEx.add(1)` must not take a double, as `numEx` would allow. This explains why there needs to be at least one extra convenience method on IntEx.

If I understand it, the suggestion is effectively to add:

```
    default NumberExpression add(final int add) {
        return this.add(Expressions.of(add));
    }
```

to NumberExpression. But users can already pass in integers using `default NumberExpression add(final double add)`. Even putting that aside, why not for all the underlying numeric types? This is addressed by my https://github.com/mongodb/mongo-java-driver/pull/1037#discussion_r1017253457 ; to elaborate, the drawback would be that for every arithmetic method taking a parameter, we would need not 0-1 but 5 overloads (and docs).

Reply:
I understand why we have `add(IntegerExpression)` on `NumberExpression` (so that we have `add` that returns `IntegerExpression`). As a result, it makes sense to introduce the `IntegerExpression.add(int)` convenience method. That's clear. I also understand that you don't want to have convenience methods for all types, and you picked a single most useful type `double`<sup>1</sup> and introduced a convenience method for this type.

> why not for all the underlying numeric types?

Because the convenience method `add(int)` was introduced on `IntegerExpression` anyway, and the same is not true for other types.

> users can already pass in integers using default NumberExpression add(final double add)

Yes, they can. For this fact to work as an argument against having `add(int)`, any `int` value must be representable perfectly as `double`, otherwise there would have been still room for users to use `NumberExpression.add(of(int))`. And indeed, any `int` value can be represented perfectly as `double`, which is something I did not initially realize.

---

<sup>1</sup> `double` can't perfectly represent all `long`/`Decimal128` values because it has only 52 bits for representing mantissa.